pkg/kubelet/status/status_manager.go

```diff
@@ -128,12 +128,35 @@ func isStatusEqual(oldStatus, status *v1.PodStatus) bool {
 	return apiequality.Semantic.DeepEqual(status, oldStatus)
 }
 
+// updateLocalPod updates the kubelet pod in the podManager if this pod isn't from an apiserver
+// The common use-case is providing a manifest-path containing Pod resources to bootstrap the Master role
+func (m *manager) updateLocalPod(uid types.UID, status versionedPodStatus) {
+	pod, podOk := m.podManager.GetPodByUID(uid)
```

Comment:
There is still a race that the pod is updated after this point but before `UpdatePod` underlying.
In that case, the pod spec might still be changed accidentally.